filelog.go

```diff
@@ -136,28 +163,42 @@ func (w *FileLogWriter) Rotate() {
 	w.rot <- true
 }
 
+// Generate the next filename for rotation
+func (w *FileLogWriter) nextFilename(filename string, suffix string) (string, error) {
+        // Attempt filename.suffix
+	fullName := fmt.Sprintf("%s.%s", filename, suffix)
+	if _, err := os.Stat(fullName); os.IsNotExist(err) {
+		// File does not exist, return it as the next filename
+		return fullName, nil
+	}
+
+	// If necessary, add integer suffix
+	for i := 1; i < 10000; i++ {
+		fullNameWithSuffix := fmt.Sprintf("%s.%s.%04d", filename, suffix, i)
+		if _, err := os.Stat(fullNameWithSuffix); os.IsNotExist(err) {
```

Comment:
@MarkRoddy Actually, the error is `permission denied` because the file isn't created until later. This section just attempts to discover if the file doesn't exist. I *think* if the directory can't be listed, `os.IsNotExist(err)` would return `true`, or at least that seems to be the behavior I'm seeing in my tests.